src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -34,139 +34,87 @@ import Button from '../../../../components/Button';
-            let shouldResetPaymentMethodData = false;
-
-            if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(this.props.bankAccountList[this.state.methodID])) {
-                shouldResetPaymentMethodData = true;
-            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(this.props.cardList[this.state.methodID])) {
-                shouldResetPaymentMethodData = true;
-            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && this.props.payPalMeData !== prevProps.payPalMeData && _.isEmpty(this.props.payPalMeData)) {
-                shouldResetPaymentMethodData = true;
-            }
-            if (shouldResetPaymentMethodData) {
-                // Close corresponding selected payment method modals which are open
-                if (this.state.shouldShowDefaultDeleteMenu) {
-                    this.hideDefaultDeleteMenu();
-                } else if (this.state.shouldShowPasswordPrompt) {
-                    this.hidePasswordPrompt();
-                }
-            }
-        }
-
-        // previously online OR currently offline, skip fetch
-        if (!prevProps.network.isOffline || this.props.network.isOffline) {
-            return;
+function BasePaymentsPage(props) {
+    const [shouldShowAddPaymentMenu, setShouldShowAddPaymentMenu] = useState(false);
+    const [shouldShowDefaultDeleteMenu, setShouldShowDefaultDeleteMenu] = useState(false);
+    const [showPassword, setShowPassword] = useState({
+        shouldShowPasswordPrompt: false,
+        passwordButtonText: '',
+    });
+    const [shouldShowLoadingSpinner, setShouldShowLoadingSpinner] = useState(false);
+    const [paymentMethod, setPaymentMethod] = useState({
+        isSelectedPaymentMethodDefault: false,
+        selectedPaymentMethod: {},
+        formattedSelectedPaymentMethod: {
+            title: '',
+        },
+        methodID: null,
+        selectedPaymentMethodType: null,
+    });
+    const [anchorPosition, setAnchorPosition] = useState({
+        anchorPositionHorizontal: 0,
+        anchorPositionVertical: 0,
+        anchorPositionTop: 0,
+        anchorPositionRight: 0,
+    });
+    const [addPaymentMethodButton, setAddPaymentMethodButton] = useState(null);
+    const [showConfirmDeleteContent, setShowConfirmDeleteContent] = useState(false);
+
+    const updateShouldShowLoadingSpinner = useCallback(() => {
+        // In order to prevent a loop, only update state of the spinner if there is a change
+        const showLoadingSpinner = props.isLoadingPaymentMethods || false;
+        if (showLoadingSpinner !== shouldShowLoadingSpinner) {
+            setShouldShowLoadingSpinner(props.isLoadingPaymentMethods && !props.network.isOffline);
         }
+    }, [props.isLoadingPaymentMethods, props.network.isOffline, shouldShowLoadingSpinner]);
```

Comment:
Why this should be a callback?

Reply:
We could keep the function block alone, call it from useEffect for network/ changes and debounced Method

Reply:
Hi, @Santhosh-Sellavel 
`updateShouldShowLoadingSpinner` and `debounceSetShouldShowLoadingSpinner` is called by here useEffect.
```
    useEffect(() => {
        // If the user was previously offline, skip debouncing showing the loader
        if (!props.network.isOffline) {
            updateShouldShowLoadingSpinner();
        } else {
            debounceSetShouldShowLoadingSpinner();
        }
    }, [props.network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);
```
As you can see, whenever isOffline was changed, useEffect calls different function.
And `updateShouldShowLoadingSpinner` has props variables like `isLoadingPaymentMethods` and `isOffline` and these variables takes part to set `state` in component.
So I thought it will be good to wrap in useCallback.

If you want to move `updateShouldShowLoadingSpinner`'s body to move to useEffect directly, I will update it.
What do you think?

Reply:
Let it be, fine as it is!